libarchive/archive_read_open_fd.c

```diff
@@ -131,15 +131,22 @@ static int64_t
 file_skip(struct archive *a, void *client_data, int64_t request)
 {
 	struct read_fd_data *mine = (struct read_fd_data *)client_data;
+#if HAVE__FSEEKI64
```

Comment:
This isn't needed here.   This module works with a file descriptor, so `lseek()` is appropriate on all platforms.  This logic is only needed in `archive_read_open_file.c` because it works with a `FILE` handle and those APIs do vary across platforms.